src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -1367,8 +1367,18 @@ private void setupCustomDeviceMapping(List<BlockDeviceMapping> deviceMappings) {
                 spotRequest.setBlockDurationMinutes(getSpotBlockReservationDuration() * 60);
             }
 
-            // Make the request for a new Spot instance
-            RequestSpotInstancesResult reqResult = ec2.requestSpotInstances(spotRequest);
+            RequestSpotInstancesResult reqResult;
+            try {
+                // Make the request for a new Spot instance
+                reqResult = ec2.requestSpotInstances(spotRequest);
```

Comment:
There seems to be some logic below for something similar could the logic be merged instead of duplicated?

Reply:
The logic is not really duplicated. In this case the client would just throw an exception, while in SlaveTemplate.java:1405 the logic is based on the status of spots requests that were created successfully. (hence the need to cancel them before falling back to on demand)